src/gsi/read_viirs.f90

```diff
@@ -0,0 +1,531 @@
+           data_mesh(25,itx) = sn                     ! snow depth
+           data_mesh(26,itx) = zz                     ! surface height
+           data_mesh(27,itx) = idomsfc + 0.001_r_kind ! dominate surface type
+           data_mesh(28,itx) = sfcr                   ! surface roughness
+           data_mesh(29,itx) = ff10                   ! ten meter wind factor
+           data_mesh(30,itx) = dlon_earth_deg         ! earth relative longitude (degrees)
+           data_mesh(31,itx) = dlat_earth_deg         ! earth relative latitude (degrees)
+           if(dval_use)then
+              data_mesh(32,itx) = val_viirs           ! weighting factor applied to super obs
+              data_mesh(33,itx) = itt                 !
+           end if
+
+           if(nst_gsi>0) then
+              data_mesh(maxinfo+1,itx) = tref         ! foundation temperature
+              data_mesh(maxinfo+2,itx) = dtw          ! dt_warm at zob
+              data_mesh(maxinfo+3,itx) = dtc          ! dt_cool at zob
+              data_mesh(maxinfo+4,itx) = tz_tr        ! d(Tz)/d(Tr)
+           endif
+
+           data_mesh(1+nreal,itx) = bufrf(2,1)
+           data_mesh(2+nreal,itx) = bmiss
+           data_mesh(3+nreal,itx) = bmiss
+           data_mesh(4+nreal,itx) = bufrf(2,2)
+           data_mesh(5+nreal,itx) = bufrf(2,3)
+
+!          write(*,'(a,I8,6F9.3)') 'viirs data : ',itx,data_mesh(7,itx),data_mesh(8,itx),data_mesh(9,itx), &
+!                     data_mesh(1+nreal,itx),data_mesh(4+nreal,itx),data_mesh(5+nreal,itx)
+
+           nrec(itx)=irec
+
+!    End of satellite read block
+
+        enddo read_loop
+     enddo read_msg
+     call closbf(lnbufr)
+
+
+
+     call combine_radobs(mype_sub,mype_root,npe_sub,mpi_comm_sub,&
+          nele,itxmax,nread,ndata_mesh,data_mesh,score_crit,nrec)
+
+     if ( nread > 0 ) then
+        write(*,'(a,a10,I3,F6.1,3I10)') 'read_viirs,jsatid,imesh,amesh,itxmax,nread,ndata_mesh :',jsatid,imesh,amesh(imesh),itxmax,nread,ndata_mesh
+     endif
+!
+!    get data_all by combining data from all thinning box sizes
+!
+     do n = 1, ndata_mesh
+        ndata = ndata + 1
+        do iele = 1, nele
+           data_all(iele,ndata) = data_mesh(iele,n)
+        enddo
+     enddo
+
+!    Deallocate arrays
+     call destroygrids
+     deallocate(data_mesh,nrec)
+
+  enddo     ! do imesh = 1, nmesh
+
+
+! Allow single task to check for bad obs, update superobs sum,
+! and write out data to scratch file for further processing.
+  if (mype_sub==mype_root.and.ndata>0) then
+     do n=1,ndata
+        do k=1,nchanl
+           if(data_all(k+nreal,n) > tbmin .and. &
+              data_all(k+nreal,n) < tbmax) nodata=nodata+1
+        end do
+     end do
+     if(dval_use .and. assim)then
+        do n=1,ndata
+           itt=nint(data_all(33,n))
+           super_val(itt)=super_val(itt)+val_viirs
+        end do
+     end if
+     write(6,'(a,a10,2I8)') 'READ_VIIRS: jsatid,mype, total number of obs info,ndata : ',jsatid,mype,ndata
```

Comment:
Is this write on line 514 for debugging purposes?  If it is, please remove this line.

Reply:
Mike,
Thanks!

I will handle what you pointed out. And just let, the reviewers and you know, it is very likely to add something to deal with the saildrone observation by modifying read_nsstbufr.f90 to this GSI code update.

Reply:
@MichaelLueken-NOAA @ADCollard  : I have update the fork master to incorporate the suggested changes by you.